From 97f2eafce46a8e86f875019ccdb1b5f7a9bef862 Mon Sep 17 00:00:00 2001
From: Elliot Saba <staticfloat@gmail.com>
Date: Tue, 20 Sep 2022 09:49:08 -0700
Subject: [PATCH] Limit initial OpenBLAS thread count

We set OpenBLAS's initial thread count to `1` to prevent runaway
allocation within OpenBLAS's initial thread startup.  LinearAlgebra will
later call `BLAS.set_num_threads()` to the actual value we require.
---
 deps/checksums/openblas                       | 184 +++++++++---------
 deps/openblas.mk                              |   7 +-
 .../openblas-default-num-threads.patch        |  57 ++++++
 stdlib/OpenBLAS_jll/Project.toml              |   2 +-
 stdlib/OpenBLAS_jll/src/OpenBLAS_jll.jl       |   5 +
 5 files changed, 161 insertions(+), 94 deletions(-)
 create mode 100644 deps/patches/openblas-default-num-threads.patch

diff --git a/deps/checksums/openblas b/deps/checksums/openblas
index 5c9194f5dd404..0dc64d447b155 100644
--- a/deps/checksums/openblas
+++ b/deps/checksums/openblas
@@ -1,94 +1,94 @@
-OpenBLAS.v0.3.21+0.aarch64-apple-darwin-libgfortran5.tar.gz/md5/a89e1eeae1d9539c21598c98da5ac91c
-OpenBLAS.v0.3.21+0.aarch64-apple-darwin-libgfortran5.tar.gz/sha512/181334363669482ac78b17ed9797ecc62ead35c07348eddd674c06252a7b36a356db48c62da77e73072df4cc21d0c25e0fb568c4dc7fe98e5db8e0f76eed7183
-OpenBLAS.v0.3.21+0.aarch64-linux-gnu-libgfortran3.tar.gz/md5/7ed3100359f9ed7da4959ecee3b4fd1e
-OpenBLAS.v0.3.21+0.aarch64-linux-gnu-libgfortran3.tar.gz/sha512/329e26dae7f2e5ba81ba2271257c03426a349b89831147458a71d91da062bd11fab1b846f77922f3bc65a9f7d3b1914f15aaa0c14f696ba7bf43b55628a5464d
-OpenBLAS.v0.3.21+0.aarch64-linux-gnu-libgfortran4.tar.gz/md5/ae71b44c62d42c246a21385d0691dcfa
-OpenBLAS.v0.3.21+0.aarch64-linux-gnu-libgfortran4.tar.gz/sha512/0598da4b6410f59a69690e6908c80724df4a8c4761194993c1b127f84418f046d8fa0d367fda8a7faed5cec2d6c57bd8872ba216e38b5418bc9ff20af27528c6
-OpenBLAS.v0.3.21+0.aarch64-linux-gnu-libgfortran5.tar.gz/md5/fff9e118d250bebd55723e77c492280c
-OpenBLAS.v0.3.21+0.aarch64-linux-gnu-libgfortran5.tar.gz/sha512/29d831f773cb6a75119c7cc2363fd72d38e32eaef6124505f8b5a1b64fa3ae7a6ffe199aae851de0893259d3bdc480aa377294688ee55d20f0da7dfc49fce747
-OpenBLAS.v0.3.21+0.aarch64-linux-musl-libgfortran3.tar.gz/md5/e93af05f98be926b3000dac3accf5f56
-OpenBLAS.v0.3.21+0.aarch64-linux-musl-libgfortran3.tar.gz/sha512/de3e9cee786d1a37dd5807aca81728d67d60fdace68aa17c69efcc7ebfe36dd3a240dea16f7cd3c5021f0f967f15f1154a3d32350f91165a9fcdd08285917196
-OpenBLAS.v0.3.21+0.aarch64-linux-musl-libgfortran4.tar.gz/md5/5a0226c6781c748a8f4d144b0ae4609b
-OpenBLAS.v0.3.21+0.aarch64-linux-musl-libgfortran4.tar.gz/sha512/83d9ff97a5651b682ee1839cf0e1aa8dcd7c2e2d32b6cadb184b8d71123649a31519516b1c7d98c329ab9902538a01ffc14ec28f95ada35ba8da77241d74c2d2
-OpenBLAS.v0.3.21+0.aarch64-linux-musl-libgfortran5.tar.gz/md5/f09c8409b0f4e5e3ee9d471538289e45
-OpenBLAS.v0.3.21+0.aarch64-linux-musl-libgfortran5.tar.gz/sha512/b41234be953779db6601d5bffe43ab9ea23bb542583b16af48fe3a3400b1e50b45d3c91152895c92f6a1f4844ac018c8003f0fd10e9473c503e70e9fc4ad11b0
-OpenBLAS.v0.3.21+0.armv6l-linux-gnueabihf-libgfortran3.tar.gz/md5/78ea013e0ba52d289c919df3d5b79946
-OpenBLAS.v0.3.21+0.armv6l-linux-gnueabihf-libgfortran3.tar.gz/sha512/50ffb9d99d283db2880e767a3ebedbdc7ca67b18782717f5085e0cfc9f6cc55bdeb112e8dca0011e31954a22272936043ca538204fc9be81cb7a0f22d6726f12
-OpenBLAS.v0.3.21+0.armv6l-linux-gnueabihf-libgfortran4.tar.gz/md5/47d016b3a50c0c9b2ed1eb5e49904169
-OpenBLAS.v0.3.21+0.armv6l-linux-gnueabihf-libgfortran4.tar.gz/sha512/d38fe6df24d658a0f54ab007ac6f9b30c0a02fbf86c0592f2e5cf5a8375b654a7428b75f74c20d97d6c953ae9998664c82800813dfa806a5f16dfc20c798c419
-OpenBLAS.v0.3.21+0.armv6l-linux-gnueabihf-libgfortran5.tar.gz/md5/17f22b32a0715f82fd48cc5d82f6512c
-OpenBLAS.v0.3.21+0.armv6l-linux-gnueabihf-libgfortran5.tar.gz/sha512/d9318a4cd232abc58907522c20ac666e6db2a92e6a25c9ddd1db0f0169be6f94aadc808014545e92e6168eaa7fa20de4235072186c48ade2e7fc672a700485ad
-OpenBLAS.v0.3.21+0.armv6l-linux-musleabihf-libgfortran3.tar.gz/md5/d5a83c8835ad8553d304bf5260b53985
-OpenBLAS.v0.3.21+0.armv6l-linux-musleabihf-libgfortran3.tar.gz/sha512/3b506824730c7269eb49e90dc11bfde2b17338ef1504da63e84562433c68586a71b022ad37de3096e06ac24e98828b48638c672789320f76cb33eda4c8e8c9eb
-OpenBLAS.v0.3.21+0.armv6l-linux-musleabihf-libgfortran4.tar.gz/md5/59cc4a5aeb63dd84c0dc12cbef7d37af
-OpenBLAS.v0.3.21+0.armv6l-linux-musleabihf-libgfortran4.tar.gz/sha512/1e9cadcf65232d4a4fba9cda0226e5a5450227e16bf2c27a3268452d5c4e4d5d1321601fd6e1a5c5d92fbc3fc4de21c92e231b5ad3b25dd71eb49d5940fcf243
-OpenBLAS.v0.3.21+0.armv6l-linux-musleabihf-libgfortran5.tar.gz/md5/c04f400d9aca82aac956e94d9fc6fc51
-OpenBLAS.v0.3.21+0.armv6l-linux-musleabihf-libgfortran5.tar.gz/sha512/8751300546ccb059fb7755e3f745e7e195cfaf90daf28b151ea2a3d540edf8910c97351d428dda196707599361a200f1a647098fdf5d7163380b4ad2b4a4f733
-OpenBLAS.v0.3.21+0.armv7l-linux-gnueabihf-libgfortran3.tar.gz/md5/f251abd323b6bc463ae4a1989670aefb
-OpenBLAS.v0.3.21+0.armv7l-linux-gnueabihf-libgfortran3.tar.gz/sha512/4f179ed09a5f5b71033d09db3894ad78d58a4429d65203225ab7a2a8c887222924910756a5fc4e3974a7cc6f9d994af287490f53cd05fe90f86c4bd4c4023b6d
-OpenBLAS.v0.3.21+0.armv7l-linux-gnueabihf-libgfortran4.tar.gz/md5/f8ffa30a958448028d1294da9d15f3b2
-OpenBLAS.v0.3.21+0.armv7l-linux-gnueabihf-libgfortran4.tar.gz/sha512/0a2d9a73be439d78b4af7c70345bdffd531d5687adeea28049adba3c8c9ab7b6ed221703f2a8aff9e7391305355a596dc9a846c84d36d1b4cdfda521f7c05e8c
-OpenBLAS.v0.3.21+0.armv7l-linux-gnueabihf-libgfortran5.tar.gz/md5/56a4aa4f3eafff290d38dc251a5966cb
-OpenBLAS.v0.3.21+0.armv7l-linux-gnueabihf-libgfortran5.tar.gz/sha512/fce03b1c1becdac66208de20e66949aba113ce2695217d34703a9ba4fd79d364cdf05424282b9d1f25ad9c315baffca3a8bd0af239f6284bd37cbdb2ec3463c6
-OpenBLAS.v0.3.21+0.armv7l-linux-musleabihf-libgfortran3.tar.gz/md5/aca7ef7f854cfe45bedbf1f6b5a97aaf
-OpenBLAS.v0.3.21+0.armv7l-linux-musleabihf-libgfortran3.tar.gz/sha512/16821728c7a0e56589b60176a06543240f2a061b437dc1cf38723dc56910c6add114a4a5e65eda051e5e88ff7b82b8613ffaf5dad7864f1208eb381159bacc8c
-OpenBLAS.v0.3.21+0.armv7l-linux-musleabihf-libgfortran4.tar.gz/md5/9f2c39eef42e5c80f56d36aee156e2b0
-OpenBLAS.v0.3.21+0.armv7l-linux-musleabihf-libgfortran4.tar.gz/sha512/441a7833e4e2d1334aeda58d87560e613244138005bc54c74af00d81c26e1e508ce874fccdcd3b3114a74f5e2a102eb520a2e4165b461861ba79fbaff81e4ae3
-OpenBLAS.v0.3.21+0.armv7l-linux-musleabihf-libgfortran5.tar.gz/md5/bf1a9f3e553d6855133b4de3ffc841ee
-OpenBLAS.v0.3.21+0.armv7l-linux-musleabihf-libgfortran5.tar.gz/sha512/2002b305ef3f3721287ed839d6d08f34127058e6295233f8280cc3ebd06d91cb326ea83f13c0158820c381fa8a2cc74ec1360a65c99bc525f492be561b15cc09
-OpenBLAS.v0.3.21+0.i686-linux-gnu-libgfortran3.tar.gz/md5/6051a0545d272bf19988e2a889531acd
-OpenBLAS.v0.3.21+0.i686-linux-gnu-libgfortran3.tar.gz/sha512/21706d2fd8dde6e1e938b0c7a53c6073d52d830d7672702d828d5606582e2c98bdb39fc7ff1fa67188942713e9d718fdf5f014812115d0d0853066c2df21f297
-OpenBLAS.v0.3.21+0.i686-linux-gnu-libgfortran4.tar.gz/md5/4fc17cff9c7a4512245ffead4d75c678
-OpenBLAS.v0.3.21+0.i686-linux-gnu-libgfortran4.tar.gz/sha512/5a28c5a587be439cd2bdf4880caf967cdec14945d26c978fa5c59ce251d5811d460bebc038808e0e8dd2eb4b6a0fdfcaacca4718e2aeb7855f466bd13d1974a7
-OpenBLAS.v0.3.21+0.i686-linux-gnu-libgfortran5.tar.gz/md5/06fa8dff91cff8ba91e2b4bc896e776c
-OpenBLAS.v0.3.21+0.i686-linux-gnu-libgfortran5.tar.gz/sha512/493110d06c4361df13ba8e0839b9410551b0bba4fe6e3cdcb53c0dff41a03b3e34ec1c2e73cd4671516631492a16b8dd140a59fa3ac71c348e670858654f2d8a
-OpenBLAS.v0.3.21+0.i686-linux-musl-libgfortran3.tar.gz/md5/1b16814a10900c96537b5bfed19e71c2
-OpenBLAS.v0.3.21+0.i686-linux-musl-libgfortran3.tar.gz/sha512/603b2a2fd92286143cb0cb573e3c085db485cf3c4f54987d255efedaea2a8a3d84b83e28f6b2db9dbf05cd31f061845a749b8402d145cc5e8cc2eb369b38e3f5
-OpenBLAS.v0.3.21+0.i686-linux-musl-libgfortran4.tar.gz/md5/20ec87e486f1154d253bc251c1ec0bce
-OpenBLAS.v0.3.21+0.i686-linux-musl-libgfortran4.tar.gz/sha512/a2d1736e0f632bddc5140ea88840113b80fedcad51bf5ea93445053eb07c1ae304a1510a85cf964d3a0e087390b8526a0df2bcd24e356b4693a41e5dfc8a671c
-OpenBLAS.v0.3.21+0.i686-linux-musl-libgfortran5.tar.gz/md5/df9f4898d550130b0c0c5cede6dd4db3
-OpenBLAS.v0.3.21+0.i686-linux-musl-libgfortran5.tar.gz/sha512/c4c3133904e7b401c5d3013d0ef38b13d7a9804fb4ba38a2c0a062f1badb4d9150214bfc2a1bf55df1299e4151d71a6dbfce7063d9d80a19fe321878c0e59309
-OpenBLAS.v0.3.21+0.i686-w64-mingw32-libgfortran3.tar.gz/md5/3d83d839f80abbd17f84631fbb8f6624
-OpenBLAS.v0.3.21+0.i686-w64-mingw32-libgfortran3.tar.gz/sha512/e51680b4b97f404f244b4d943148f506f84a21f0f59f4d41a3a0cf81d545658d9cc59592a2538c4c077606fc1e6f87eda23063e49828f066e743205c5e6aee8e
-OpenBLAS.v0.3.21+0.i686-w64-mingw32-libgfortran4.tar.gz/md5/2ca3ebd61038a5e422a946ede3d17535
-OpenBLAS.v0.3.21+0.i686-w64-mingw32-libgfortran4.tar.gz/sha512/c692b7c00aa0eda4a3fa989f84c797126b1596e13514117ad898166f941f040df67826ba33d0af93673c7a0b478fe4392f9a53d7859b7371878b6608dcb0556b
-OpenBLAS.v0.3.21+0.i686-w64-mingw32-libgfortran5.tar.gz/md5/55995848e4bc9ef739e8ba17962787d3
-OpenBLAS.v0.3.21+0.i686-w64-mingw32-libgfortran5.tar.gz/sha512/e274dcfc6593786ac4c32c3f49ec69ab3a0c7278c67bbd04373d4845bff2dfaf17300d4a71e48ebd508e299fa629190ffe70ce074a68e83bac0eafa51f4db2a0
-OpenBLAS.v0.3.21+0.powerpc64le-linux-gnu-libgfortran3.tar.gz/md5/2a3d5f0240a27cf1617d9d4abba6df78
-OpenBLAS.v0.3.21+0.powerpc64le-linux-gnu-libgfortran3.tar.gz/sha512/489128c884a0c2b9eb5145b18c68f9c7de9a9cc4131f4635689dc67765f87bec852b0547ebea4ecfdad4eec38063aabe8f6c3e96e5856e08c0c26130c2f11897
-OpenBLAS.v0.3.21+0.powerpc64le-linux-gnu-libgfortran4.tar.gz/md5/bb3501965c26519ecf30830465b12672
-OpenBLAS.v0.3.21+0.powerpc64le-linux-gnu-libgfortran4.tar.gz/sha512/bcc78c10f7aba973f1159443e8d408465833ef43530c1c2514715db5a1bb84c0f48510c12b2ac1211b2777328ec682e01ab668818025651f00db7ca747f5674e
-OpenBLAS.v0.3.21+0.powerpc64le-linux-gnu-libgfortran5.tar.gz/md5/dc6192b5130e114b3cb1183690c7e398
-OpenBLAS.v0.3.21+0.powerpc64le-linux-gnu-libgfortran5.tar.gz/sha512/95d69ace7b145c02acbe13c52d1e7835fdb1e8788c0b03f0f967d88575b322988e4f4acef3b6ad3e505c895f8d19566b8eb9add02f0250cf2d4a14c9f1428f3f
-OpenBLAS.v0.3.21+0.x86_64-apple-darwin-libgfortran3.tar.gz/md5/27a9117002f96c41c7907be0475a8d86
-OpenBLAS.v0.3.21+0.x86_64-apple-darwin-libgfortran3.tar.gz/sha512/2247f3691cc552f65a353412c46a76c1ac4b4d994a5725044ba300f0944f15b37144ceff438d77022b441c25eaf530709a4d3ed4523b97d292991b6407a72970
-OpenBLAS.v0.3.21+0.x86_64-apple-darwin-libgfortran4.tar.gz/md5/30dea9ca8b658ff6a9db9245d8ad7500
-OpenBLAS.v0.3.21+0.x86_64-apple-darwin-libgfortran4.tar.gz/sha512/3289e766bfad456b35efae6d341a77698d4d36908ac8d802f47777feed5eef224fde1cb4799b5bd4e8e216c28c78ab4407b92906ddac0bdd1cfb674136c69aaa
-OpenBLAS.v0.3.21+0.x86_64-apple-darwin-libgfortran5.tar.gz/md5/ed820c420a67b32df0a9f34760ce605c
-OpenBLAS.v0.3.21+0.x86_64-apple-darwin-libgfortran5.tar.gz/sha512/63f43eb0b1c037849fd5addda66d818c0b8188f9758391cd9929e78593c69ec61729be0efce6a9f943ebac178c634cdddffe172ad681ad1c4717949b075a1de7
-OpenBLAS.v0.3.21+0.x86_64-linux-gnu-libgfortran3.tar.gz/md5/a5393eb8e1799b6c089a82d8dde39fb0
-OpenBLAS.v0.3.21+0.x86_64-linux-gnu-libgfortran3.tar.gz/sha512/8ce9ad2f55f131592a87789ec6a824cbe1d23c3be32fb2ba59e107045f75c34684b85d3bab2913923f5a19414a072b5e855c86fddd44a4319a9b5e7b28d5c169
-OpenBLAS.v0.3.21+0.x86_64-linux-gnu-libgfortran4.tar.gz/md5/652aa333440219a4ec17d94dd4e6d358
-OpenBLAS.v0.3.21+0.x86_64-linux-gnu-libgfortran4.tar.gz/sha512/89d7740b462d4216e920dcd5c7867520b2f49c3cb74bd8424efd287927c92a08492c9fa413506248207f9532c7bb9ea2af587a4f70d7db8ea42ac1bc144e8a12
-OpenBLAS.v0.3.21+0.x86_64-linux-gnu-libgfortran5.tar.gz/md5/9e0831544d02a39565a2d0714b1e121a
-OpenBLAS.v0.3.21+0.x86_64-linux-gnu-libgfortran5.tar.gz/sha512/9e57a08a09d5fc47d881f9f7ed2e52fbdc7301908cf1be384fe767e6b7771a5980827417049dd37df4d71a861b2cf2a05f25df892b15ed03458910b0bc53101a
-OpenBLAS.v0.3.21+0.x86_64-linux-musl-libgfortran3.tar.gz/md5/dde15d3a2f26601cd7ac0a803efbe503
-OpenBLAS.v0.3.21+0.x86_64-linux-musl-libgfortran3.tar.gz/sha512/fa48e51a59b6fb213f88ce8b2778ca5eef73f0721a5c71e27cd8952a34a5003d69710571deb5c0c06526fa8016cfdacabdc2b343342ad0d1e523903fa94a8d42
-OpenBLAS.v0.3.21+0.x86_64-linux-musl-libgfortran4.tar.gz/md5/55b80d13104f4ddc9eefa0424d71330b
-OpenBLAS.v0.3.21+0.x86_64-linux-musl-libgfortran4.tar.gz/sha512/fcd816e4dcd767963ae555846cee846c19f0b7d666042d59757eb2eebe115d30af60072c134c3025049712520705dbe9d2862a1f07c955780f9a0de69e6e00b5
-OpenBLAS.v0.3.21+0.x86_64-linux-musl-libgfortran5.tar.gz/md5/e22e46b304962a1edb550e19914cc5e6
-OpenBLAS.v0.3.21+0.x86_64-linux-musl-libgfortran5.tar.gz/sha512/38f8ce1baee4082fd2432dbc1905fd03d8efbcc378aefc9312e90b6054341717ea46bc4d33f9f517e67af0fca2da55b5c5b112850e6071ba18753a4936d78da2
-OpenBLAS.v0.3.21+0.x86_64-unknown-freebsd-libgfortran3.tar.gz/md5/b8b7a9f9aff94b154e8479a84d7abe4b
-OpenBLAS.v0.3.21+0.x86_64-unknown-freebsd-libgfortran3.tar.gz/sha512/afe4c45d4bf4e38fdbbf00b9e86a65244aadaa2b74e59f9a43f1860c130f721bba2f36186519b2573ff0819bd2b30414cc23800634847db2ecd2107f985495ad
-OpenBLAS.v0.3.21+0.x86_64-unknown-freebsd-libgfortran4.tar.gz/md5/8fceea42a8524fef29a54b88ea0a721b
-OpenBLAS.v0.3.21+0.x86_64-unknown-freebsd-libgfortran4.tar.gz/sha512/e8195597057ab6de0aa5978b4d13b3248ac6acde3f86fc55d9e1c76ec39d464fc2eefea1096cfb5dffbd623f47b06be163c4c74981d2eb13387bc8499b9053fe
-OpenBLAS.v0.3.21+0.x86_64-unknown-freebsd-libgfortran5.tar.gz/md5/6f88d96c05663b4aeb81ba8a63882066
-OpenBLAS.v0.3.21+0.x86_64-unknown-freebsd-libgfortran5.tar.gz/sha512/f1f516d8d16a2f86bfb26129e0530146de3a4bcb62abcd2c7b9bf64cc09c069e5eeb66658b1cc0cdcc665de98246ad8ed20a7d8b132fe0f0e4d0651d3b4516d4
-OpenBLAS.v0.3.21+0.x86_64-w64-mingw32-libgfortran3.tar.gz/md5/4fb99984ec612a090b294c6b349a7cdb
-OpenBLAS.v0.3.21+0.x86_64-w64-mingw32-libgfortran3.tar.gz/sha512/81bf55c6398a6cf4a61add084332e7cb79e6d550737641f6c0bc77aa61bd8187603a6100b78c2ef80109c3c5b21f7ba618a4780a5b68e5168a461af521f26c52
-OpenBLAS.v0.3.21+0.x86_64-w64-mingw32-libgfortran4.tar.gz/md5/a1a2c3623d583ab995ac86df07ab73bb
-OpenBLAS.v0.3.21+0.x86_64-w64-mingw32-libgfortran4.tar.gz/sha512/ec5fab349607862c9e0badaa1fd433e057ac7d056008af683bbb37bf43fef5322e598cd71a6d9c3dd55ef857b39ca634e64572e9ae6e263022dc7f89083f9bca
-OpenBLAS.v0.3.21+0.x86_64-w64-mingw32-libgfortran5.tar.gz/md5/05ef0ede7d565247a936c3110c25c83c
-OpenBLAS.v0.3.21+0.x86_64-w64-mingw32-libgfortran5.tar.gz/sha512/34d2812bc0c6605558cbd6244c41d0805fc9a943cd91a74994bcacdd5ff19612eac8751832e3ee761089a853cf16069e67e13507ca75bbe4b7dc4517e41515e0
+OpenBLAS.v0.3.21+3.aarch64-apple-darwin-libgfortran5.tar.gz/md5/fa267987c77c4f20010456e3ba08e91e
+OpenBLAS.v0.3.21+3.aarch64-apple-darwin-libgfortran5.tar.gz/sha512/b3a018b21cc60a91b0439f34ddbf57064d347a10bf8f9a1b813f7af93b1a2fd2b552ebe30baa515fcc1583c4b9376c106881fa742d409285a06a06f2b4f2f6da
+OpenBLAS.v0.3.21+3.aarch64-linux-gnu-libgfortran3.tar.gz/md5/eff07fb30c3e35d26fa3a7d0fc837c20
+OpenBLAS.v0.3.21+3.aarch64-linux-gnu-libgfortran3.tar.gz/sha512/70000c634af84974e96ac13481ba729c4668342cd06cb98032c1ad4d23890bafc7f9c2692e8ebae7127d88fc27553e382b80edb078f1a1ba6c13535f56535685
+OpenBLAS.v0.3.21+3.aarch64-linux-gnu-libgfortran4.tar.gz/md5/0015556833727545b248985acf9a79dd
+OpenBLAS.v0.3.21+3.aarch64-linux-gnu-libgfortran4.tar.gz/sha512/5b0007054c67629b86520298b462b07908c57a0c42871db99ce834673f7f2164e0cea0b952c4dcf2678cbf75354bcf0a8feee37266ddd41ef2c33a848666b00f
+OpenBLAS.v0.3.21+3.aarch64-linux-gnu-libgfortran5.tar.gz/md5/1888c11bfc930840fe57d29ac4fb1dbd
+OpenBLAS.v0.3.21+3.aarch64-linux-gnu-libgfortran5.tar.gz/sha512/571f985d34ee403f9af09a143c695c4c626f32b363ae2439d8ca4057041fc966217c6d912089ad0c675e573253cd9327099e71f293cbe5f06be4e077da9f48ea
+OpenBLAS.v0.3.21+3.aarch64-linux-musl-libgfortran3.tar.gz/md5/ec3bb6b7d998e37b52ae61439f4e4c58
+OpenBLAS.v0.3.21+3.aarch64-linux-musl-libgfortran3.tar.gz/sha512/7ad0bbfa194f7c8c30533d81a15c4229d300ed1c108af975f381a7b644c1da377b11cff60b60ee193152b91e0d29967f79d9b32d44b54d9b2230a768427ab28a
+OpenBLAS.v0.3.21+3.aarch64-linux-musl-libgfortran4.tar.gz/md5/45bffcba3c88e111d903b95d503bead6
+OpenBLAS.v0.3.21+3.aarch64-linux-musl-libgfortran4.tar.gz/sha512/be76750ff245ad91d5e6fad268619ec8732ad1fc1236e02c8e4ff65ecbf5502fa1cae89b026e0414dfe8ec38d002acf79d18f4ae8a320da6dedb3760db969c04
+OpenBLAS.v0.3.21+3.aarch64-linux-musl-libgfortran5.tar.gz/md5/6932c7351aed13bfc25ba56a283595d5
+OpenBLAS.v0.3.21+3.aarch64-linux-musl-libgfortran5.tar.gz/sha512/2df1fd87da29d1ee85bc3208e3435756bfe615df49385072a2a18e15ba54fba754e94be0fdce9f7c370af5ff746b5841beda9a3f5f011b8472b2b579ca2eded5
+OpenBLAS.v0.3.21+3.armv6l-linux-gnueabihf-libgfortran3.tar.gz/md5/367657c9e84b6016d3b835d98af3dd2a
+OpenBLAS.v0.3.21+3.armv6l-linux-gnueabihf-libgfortran3.tar.gz/sha512/3a660c0096f086742e2bac219db732244f35bf527537302e8590ea6e6a369438268ebc479a67096e0ac872f5792f149c6097c64a8afb2624e09687fa4f3bf023
+OpenBLAS.v0.3.21+3.armv6l-linux-gnueabihf-libgfortran4.tar.gz/md5/86309e18c22fc5fa5e437bc5b8814f28
+OpenBLAS.v0.3.21+3.armv6l-linux-gnueabihf-libgfortran4.tar.gz/sha512/ca51805940efadb27fcdd845551f57a936fbdfbc3caf71fb04eb70360686b75ec5eaf8957e860622d5cbfa4305edacdcfd49bbb48134cd05b96b499faa8e2fd4
+OpenBLAS.v0.3.21+3.armv6l-linux-gnueabihf-libgfortran5.tar.gz/md5/993d99064baa38498d0c40829bc0899a
+OpenBLAS.v0.3.21+3.armv6l-linux-gnueabihf-libgfortran5.tar.gz/sha512/267ebe796afa475a41bb57d6eed1397a5b434945446cd2e612024218fa93f9342bcc4fb8cee0957422aa31ee89c77fe4b07f3f573eb01b6fad0d52d859c7df6c
+OpenBLAS.v0.3.21+3.armv6l-linux-musleabihf-libgfortran3.tar.gz/md5/243ee32320336ada3524545164ba1fd3
+OpenBLAS.v0.3.21+3.armv6l-linux-musleabihf-libgfortran3.tar.gz/sha512/2e6b95d82004b8e411441d13e90dd39e39b57288f3502d077daf811709ca1f2eab10fed66648de7cbd0ee37bebb9eef46bd5f476e9ff942f1110b4cde337cea6
+OpenBLAS.v0.3.21+3.armv6l-linux-musleabihf-libgfortran4.tar.gz/md5/586032d64e693a46dfe7ae5f56cc6bb3
+OpenBLAS.v0.3.21+3.armv6l-linux-musleabihf-libgfortran4.tar.gz/sha512/1bc53988dcb22dc82d434a151eb74eadca00ffe3822575497a10229fda967c333828137e76a4afbcc8576ac9261f492ccb4e1e70eb22da977a189b39a72bde63
+OpenBLAS.v0.3.21+3.armv6l-linux-musleabihf-libgfortran5.tar.gz/md5/a6b2acfa27a2cf042f228e3f79288161
+OpenBLAS.v0.3.21+3.armv6l-linux-musleabihf-libgfortran5.tar.gz/sha512/e4b04da8a2002946ca674e191851c506723a91837232025c9e23115348df3187b041725d2406a592e48a595aa3fbe8ff9da9ae70ad8d366e4c118fdba52c9411
+OpenBLAS.v0.3.21+3.armv7l-linux-gnueabihf-libgfortran3.tar.gz/md5/367657c9e84b6016d3b835d98af3dd2a
+OpenBLAS.v0.3.21+3.armv7l-linux-gnueabihf-libgfortran3.tar.gz/sha512/3a660c0096f086742e2bac219db732244f35bf527537302e8590ea6e6a369438268ebc479a67096e0ac872f5792f149c6097c64a8afb2624e09687fa4f3bf023
+OpenBLAS.v0.3.21+3.armv7l-linux-gnueabihf-libgfortran4.tar.gz/md5/86309e18c22fc5fa5e437bc5b8814f28
+OpenBLAS.v0.3.21+3.armv7l-linux-gnueabihf-libgfortran4.tar.gz/sha512/ca51805940efadb27fcdd845551f57a936fbdfbc3caf71fb04eb70360686b75ec5eaf8957e860622d5cbfa4305edacdcfd49bbb48134cd05b96b499faa8e2fd4
+OpenBLAS.v0.3.21+3.armv7l-linux-gnueabihf-libgfortran5.tar.gz/md5/993d99064baa38498d0c40829bc0899a
+OpenBLAS.v0.3.21+3.armv7l-linux-gnueabihf-libgfortran5.tar.gz/sha512/267ebe796afa475a41bb57d6eed1397a5b434945446cd2e612024218fa93f9342bcc4fb8cee0957422aa31ee89c77fe4b07f3f573eb01b6fad0d52d859c7df6c
+OpenBLAS.v0.3.21+3.armv7l-linux-musleabihf-libgfortran3.tar.gz/md5/243ee32320336ada3524545164ba1fd3
+OpenBLAS.v0.3.21+3.armv7l-linux-musleabihf-libgfortran3.tar.gz/sha512/2e6b95d82004b8e411441d13e90dd39e39b57288f3502d077daf811709ca1f2eab10fed66648de7cbd0ee37bebb9eef46bd5f476e9ff942f1110b4cde337cea6
+OpenBLAS.v0.3.21+3.armv7l-linux-musleabihf-libgfortran4.tar.gz/md5/586032d64e693a46dfe7ae5f56cc6bb3
+OpenBLAS.v0.3.21+3.armv7l-linux-musleabihf-libgfortran4.tar.gz/sha512/1bc53988dcb22dc82d434a151eb74eadca00ffe3822575497a10229fda967c333828137e76a4afbcc8576ac9261f492ccb4e1e70eb22da977a189b39a72bde63
+OpenBLAS.v0.3.21+3.armv7l-linux-musleabihf-libgfortran5.tar.gz/md5/a6b2acfa27a2cf042f228e3f79288161
+OpenBLAS.v0.3.21+3.armv7l-linux-musleabihf-libgfortran5.tar.gz/sha512/e4b04da8a2002946ca674e191851c506723a91837232025c9e23115348df3187b041725d2406a592e48a595aa3fbe8ff9da9ae70ad8d366e4c118fdba52c9411
+OpenBLAS.v0.3.21+3.i686-linux-gnu-libgfortran3.tar.gz/md5/e752b2e67a151ea7cfb76c9b15b687e1
+OpenBLAS.v0.3.21+3.i686-linux-gnu-libgfortran3.tar.gz/sha512/230a5fade6d4a205d932362ce40b5455229247ed279e6a0d7105d6207c28d699094c227216aad267aa053878b914043284c7dd9d1c2e4d26d0978efc9071bb48
+OpenBLAS.v0.3.21+3.i686-linux-gnu-libgfortran4.tar.gz/md5/478d429202eb287a840df5bbf191b878
+OpenBLAS.v0.3.21+3.i686-linux-gnu-libgfortran4.tar.gz/sha512/31f98104f1355c01aa86f2fb01165fdf93035319285987d93045518fc4ecc84b59a911369169fc534a4ab3b18c27fe49dc80125263d73ad7d265e6e3913d25a4
+OpenBLAS.v0.3.21+3.i686-linux-gnu-libgfortran5.tar.gz/md5/c8821f59bc938244b2bdc46278c041a5
+OpenBLAS.v0.3.21+3.i686-linux-gnu-libgfortran5.tar.gz/sha512/475413dda79b2263a19eeb657f2c31e36f2f915a702327b3331d294c8783a4f6f4f31b672edfed14cdbdbd2b4f5bf36273f70fa2dd1ec5611925f7a55971780f
+OpenBLAS.v0.3.21+3.i686-linux-musl-libgfortran3.tar.gz/md5/9ba93d57e2cf8695699b881715f32f2c
+OpenBLAS.v0.3.21+3.i686-linux-musl-libgfortran3.tar.gz/sha512/eb0ea7be9853ab1c220545f628cea4047deacf1fd704c889950f7aeafb58dc75e6cccd1b0c85c30ca12007cce05773382bf4a944f61aa9a0ed0f51621b45fc64
+OpenBLAS.v0.3.21+3.i686-linux-musl-libgfortran4.tar.gz/md5/a4f7aa370d7023f0b062a255a48ff81f
+OpenBLAS.v0.3.21+3.i686-linux-musl-libgfortran4.tar.gz/sha512/cc8fd2dd011d4007ea5bd99d909776ca2a3700f6457a92cb43f77684b0dfa5a13f808e03db142647766c385e2bbd4db97f90701f286f5cb04276153d8f6a94fa
+OpenBLAS.v0.3.21+3.i686-linux-musl-libgfortran5.tar.gz/md5/5b33f48568c3347a98a72d44786f14c7
+OpenBLAS.v0.3.21+3.i686-linux-musl-libgfortran5.tar.gz/sha512/684ac712c035b14ec1b9b1f0ebad1b813b19155852b7b2af75e5956533d8b90e09bc9652a2beb908d2a1c7e2d22fccbf5479aab10d95528aa173d603f60f6135
+OpenBLAS.v0.3.21+3.i686-w64-mingw32-libgfortran3.tar.gz/md5/713248c80e4fcadc14b2b47f7c814df8
+OpenBLAS.v0.3.21+3.i686-w64-mingw32-libgfortran3.tar.gz/sha512/bb954e6b3b2b45cd08c759626eee7f7de58addb9c7c95f688636ee792c759cacb81a026be9b2c8a7049efa42b2a8833b189dd881fb88d20e760f716ee13defd0
+OpenBLAS.v0.3.21+3.i686-w64-mingw32-libgfortran4.tar.gz/md5/4f2de55ace467aece520ee9c3720b7dc
+OpenBLAS.v0.3.21+3.i686-w64-mingw32-libgfortran4.tar.gz/sha512/3012a3b940b8a427258a669620a983d8b701d26b6a53c7b699cccb03278e01266097f05b3679422d5377435e16d721f4f56473f5a91f3ad7ec6ebc14c1125a8b
+OpenBLAS.v0.3.21+3.i686-w64-mingw32-libgfortran5.tar.gz/md5/f09e00c1f60f8fe39da3ddbbf992426e
+OpenBLAS.v0.3.21+3.i686-w64-mingw32-libgfortran5.tar.gz/sha512/06a9eabade786599566fc097e350f01d592e90bb6e2b6635a4c49e85ac5df668fe1e1a855ccdc5184f0f588f1e3e2d2ec2e3ff310cb5640ed16b56908a1e0a99
+OpenBLAS.v0.3.21+3.powerpc64le-linux-gnu-libgfortran3.tar.gz/md5/b9624b25a842386007389532a9f7e573
+OpenBLAS.v0.3.21+3.powerpc64le-linux-gnu-libgfortran3.tar.gz/sha512/fca2bd22e9aa7e9a471ad72cc8d15ce204e54e4f1df052bc3025a94beacf138a6f853585c77b884517316c26a24b8c40d9e14f91832b9d88eb022723bd7543f6
+OpenBLAS.v0.3.21+3.powerpc64le-linux-gnu-libgfortran4.tar.gz/md5/13f3ff6b22b646695aa4877ba1e0de9a
+OpenBLAS.v0.3.21+3.powerpc64le-linux-gnu-libgfortran4.tar.gz/sha512/00165490fc91537034dae93d3085f1e3b9523f63398c7610753b39a6b75fd421c0871b2be38786fb016b3575353287c60d59f8b84c72b8afcecc7464213932ad
+OpenBLAS.v0.3.21+3.powerpc64le-linux-gnu-libgfortran5.tar.gz/md5/e05e8c3cbd828e268718f95eb6e39d0a
+OpenBLAS.v0.3.21+3.powerpc64le-linux-gnu-libgfortran5.tar.gz/sha512/0bfbc85919bf31821cdf80db700937f339d11b5e30d5b51731e0c3dc59f2025c75b80782dddf4347f48cdd2e169dd41ec314a655a26047cf9306ac34c9005e66
+OpenBLAS.v0.3.21+3.x86_64-apple-darwin-libgfortran3.tar.gz/md5/92b5c555c5a892d449521ab9a440b88f
+OpenBLAS.v0.3.21+3.x86_64-apple-darwin-libgfortran3.tar.gz/sha512/f4144071d254030e18edf055db5fc057a22abfabfd6700cc8ea7febb1e87aab5b174945adb2ea663394d3cf3bc85e750beb5562f416b7e030f1ce84730a10c9c
+OpenBLAS.v0.3.21+3.x86_64-apple-darwin-libgfortran4.tar.gz/md5/fad5eafc02f87648dea8904b47966603
+OpenBLAS.v0.3.21+3.x86_64-apple-darwin-libgfortran4.tar.gz/sha512/1c8d2364f778814af6e211a43be40d03d72b0063847c876768f37656e92a7dac67a18a118a62af2d7acdb1db16c6e415d2b5bbc2abd574ce6ec4b7c78d98c6da
+OpenBLAS.v0.3.21+3.x86_64-apple-darwin-libgfortran5.tar.gz/md5/d0bb6534cb172696e3057dbf7c8c7a51
+OpenBLAS.v0.3.21+3.x86_64-apple-darwin-libgfortran5.tar.gz/sha512/1ced0e418ebc43fd138cf95ecfb67513cffbeee37a5a3def7c9a72534bc43400025441e77d9c744d9001891886da4a69f9089e495086386a44f8f4c57e5b81f3
+OpenBLAS.v0.3.21+3.x86_64-linux-gnu-libgfortran3.tar.gz/md5/9c7563a8df3c1d27e4db9f8e5288fc45
+OpenBLAS.v0.3.21+3.x86_64-linux-gnu-libgfortran3.tar.gz/sha512/c6f81ec1da096f6e27e6179235042f0558bc5e3ade27a230bafb2f121b48bc96d7d9e0c45b0e4dc8fee10257913eccabcfcaf67d515e0ba4f373f1ca4869773c
+OpenBLAS.v0.3.21+3.x86_64-linux-gnu-libgfortran4.tar.gz/md5/206f6302a757a1a30ab6716b6e508741
+OpenBLAS.v0.3.21+3.x86_64-linux-gnu-libgfortran4.tar.gz/sha512/35adbec09c4a636131a757f4b51d7b458f23e7b7118194be63a116746b2e1e7a15990bd76e8ecbbe2a6010cb6e54bca3a296f02d83b27c6394b42bdffa041653
+OpenBLAS.v0.3.21+3.x86_64-linux-gnu-libgfortran5.tar.gz/md5/9aa48579f0c50500796de228f00909d8
+OpenBLAS.v0.3.21+3.x86_64-linux-gnu-libgfortran5.tar.gz/sha512/ef1b1578259b055a23df4e57bb4bf2c1ef94e4d0d1e63eda622db291256ef8cde8694befcdc83b507cf6c3f47db142b51e6bac614ec32bae92ae576ddd6e2f15
+OpenBLAS.v0.3.21+3.x86_64-linux-musl-libgfortran3.tar.gz/md5/ba5ac7e48691965b3d5045785d37618e
+OpenBLAS.v0.3.21+3.x86_64-linux-musl-libgfortran3.tar.gz/sha512/0e37f806faa5bc5cd8e5096efbbbef458720c267ed69028e66ced3269baac53c9e0a92ab728ceab499f6aed3edbcbca6c2b1caaa6e145c386bafb57657a6c353
+OpenBLAS.v0.3.21+3.x86_64-linux-musl-libgfortran4.tar.gz/md5/e9b1ab8fcd6b62b92db6e064054a02ea
+OpenBLAS.v0.3.21+3.x86_64-linux-musl-libgfortran4.tar.gz/sha512/75706214b660df35daa3865cc267087ca9aecd04b658d5b9d867d15bad901f00677766c04d7701b45d5238aaf4ed09fafff0ca07e9f501dded9479eea26a5112
+OpenBLAS.v0.3.21+3.x86_64-linux-musl-libgfortran5.tar.gz/md5/d86ecacd9904ef70fe17b762e70986e7
+OpenBLAS.v0.3.21+3.x86_64-linux-musl-libgfortran5.tar.gz/sha512/12edb53c88d7deb4585092ca9d9e3c0578d0ca0e91b388c2eddf84dc080c971dba122195b750e2aa4c3a532a7df2c9da7b56b565f3c2002282621cc63425954e
+OpenBLAS.v0.3.21+3.x86_64-unknown-freebsd-libgfortran3.tar.gz/md5/494fccd04e83345dc20bd753319b8ed3
+OpenBLAS.v0.3.21+3.x86_64-unknown-freebsd-libgfortran3.tar.gz/sha512/3000b9ec7b66e09597759570eb77645dec1e82a0b4f56ab9439f2fa3e55bbc2863cc61cbbe77752415fd5926010b1647fffedb8dacaa77b87f815d98bb47d86b
+OpenBLAS.v0.3.21+3.x86_64-unknown-freebsd-libgfortran4.tar.gz/md5/df1133b14f2561b02f4f2e52106635ed
+OpenBLAS.v0.3.21+3.x86_64-unknown-freebsd-libgfortran4.tar.gz/sha512/ed2c18a9f0768412128dd2e370d85868e536cf3ca4d0d03bc626e4690ba81965c540f08d00b46208a7969bd07b9848e6ff8a14e91a4e927d2b44e9ba5b374e8c
+OpenBLAS.v0.3.21+3.x86_64-unknown-freebsd-libgfortran5.tar.gz/md5/0ed230111e50d49609d631a2f205dff5
+OpenBLAS.v0.3.21+3.x86_64-unknown-freebsd-libgfortran5.tar.gz/sha512/1940a500b25f1c2613d2f2ab6533309991808622082a7910985fc0573d41358c5c9311c9bb65a00d5508e7d9e4612773e97adb860cba2c2f4f3957782d695f21
+OpenBLAS.v0.3.21+3.x86_64-w64-mingw32-libgfortran3.tar.gz/md5/6a154f9fb8802ca07dafe968a31b2ed3
+OpenBLAS.v0.3.21+3.x86_64-w64-mingw32-libgfortran3.tar.gz/sha512/c969d3b6485d81a07be5145c79af7e15b39740a47d2f7c31cf1f1d75b7b8a845d53b3291aa952d23116b76fa02eeedd44ab093fa8d57138c7034bf0db389812a
+OpenBLAS.v0.3.21+3.x86_64-w64-mingw32-libgfortran4.tar.gz/md5/7b6a1bd651ce8ead09315e884e0e1699
+OpenBLAS.v0.3.21+3.x86_64-w64-mingw32-libgfortran4.tar.gz/sha512/145f8f18f8cb4caa73708b15d1facb79c93d62e84886ecff586e429650175016387aa8c01ab87e67bbacce4f942634012436359dbbfefef11fc00056173249ce
+OpenBLAS.v0.3.21+3.x86_64-w64-mingw32-libgfortran5.tar.gz/md5/30186ff3504dff0a00882c0173a94c21
+OpenBLAS.v0.3.21+3.x86_64-w64-mingw32-libgfortran5.tar.gz/sha512/92d445cd96d2b527c9bb1201f5cf7e4c17bc81805ce32a59a732498f30d9096cef74d5d9d2b559770281f9771789fdb1571ae8bca4e7b2466805eb33040fd504
 openblas-b89fb708caa5a5a32de8f4306c4ff132e0228e9a.tar.gz/md5/716ebe95d4b491253cdde8308b8adb83
 openblas-b89fb708caa5a5a32de8f4306c4ff132e0228e9a.tar.gz/sha512/00e7bde49525c2c28bf07b47290e00b53bff446be63f09e90c51724c6350e5ddc90f5a071ae6de057b3fbb107060e70bf16683fcefcf48ae37ba1d0758be553b
diff --git a/deps/openblas.mk b/deps/openblas.mk
index f949143f393b1..6e3c4f8cb0fab 100644
--- a/deps/openblas.mk
+++ b/deps/openblas.mk
@@ -100,7 +100,12 @@ $(BUILDDIR)/$(OPENBLAS_SRC_DIR)/neoverse-generic-kernels.patch-applied: $(BUILDD
 		patch -p1 -f < $(SRCDIR)/patches/neoverse-generic-kernels.patch
 	echo 1 > $@
 
-$(BUILDDIR)/$(OPENBLAS_SRC_DIR)/build-configured: $(BUILDDIR)/$(OPENBLAS_SRC_DIR)/neoverse-generic-kernels.patch-applied
+$(BUILDDIR)/$(OPENBLAS_SRC_DIR)/openblas-default-num-threads.patch-applied: $(BUILDDIR)/$(OPENBLAS_SRC_DIR)/neoverse-generic-kernels.patch-applied
+	cd $(BUILDDIR)/$(OPENBLAS_SRC_DIR) && \
+		patch -p1 -f < $(SRCDIR)/patches/openblas-default-num-threads.patch
+	echo 1 > $@
+
+$(BUILDDIR)/$(OPENBLAS_SRC_DIR)/build-configured: $(BUILDDIR)/$(OPENBLAS_SRC_DIR)/openblas-default-num-threads.patch-applied
 	echo 1 > $@
 
 $(BUILDDIR)/$(OPENBLAS_SRC_DIR)/build-compiled: $(BUILDDIR)/$(OPENBLAS_SRC_DIR)/build-configured
diff --git a/deps/patches/openblas-default-num-threads.patch b/deps/patches/openblas-default-num-threads.patch
new file mode 100644
index 0000000000000..6580e9a41d1eb
--- /dev/null
+++ b/deps/patches/openblas-default-num-threads.patch
@@ -0,0 +1,57 @@
+commit d2ce93179f6747380488db2a56102dab6fde18ca
+Author: Elliot Saba <staticfloat@gmail.com>
+Date:   Thu Sep 22 10:38:36 2022 -0700
+
+    Add `OPENBLAS_DEFAULT_NUM_THREADS`
+    
+    This allows Julia to set a default number of threads (usually `1`) to be
+    used when no other thread counts are specified [0], to short-circuit the
+    default OpenBLAS thread initialization routine that spins up a different
+    number of threads than Julia would otherwise choose.
+    
+    The reason to add a new environment variable is that we want to be able
+    to configure OpenBLAS to avoid performing its initial memory
+    allocation/thread startup, as that can consume significant amounts of
+    memory, but we still want to be sensitive to legacy codebases that set
+    things like `OMP_NUM_THREADS` or `GOTOBLAS_NUM_THREADS`.  Creating a new
+    environment variable that is openblas-specific and is not already
+    publicly used to control the overall number of threads of programs like
+    Julia seems to be the best way forward.
+    
+    [0] https://github.com/JuliaLang/julia/pull/46844
+
+diff --git a/driver/others/init.c b/driver/others/init.c
+index cc3145a6..cd10e8d3 100644
+--- a/driver/others/init.c
++++ b/driver/others/init.c
+@@ -823,6 +823,8 @@ void gotoblas_affinity_init(void) {
+ 
+   if (numprocs == 0) numprocs = readenv_atoi("OMP_NUM_THREADS");
+ 
++  if (numprocs == 0) numprocs = readenv_atoi("OPENBLAS_DEFAULT_NUM_THREADS");
++
+   numnodes = 1;
+ 
+   if (numprocs == 1) {
+diff --git a/driver/others/openblas_env.c b/driver/others/openblas_env.c
+index ef91a08e..35b2270d 100644
+--- a/driver/others/openblas_env.c
++++ b/driver/others/openblas_env.c
+@@ -67,10 +67,16 @@ void openblas_read_env() {
+   openblas_env_thread_timeout=(unsigned int)ret;
+ 
+   ret=0;
+-  if (readenv(p,"OPENBLAS_NUM_THREADS")) ret = atoi(p);
++  if (readenv(p,"OPENBLAS_DEFAULT_NUM_THREADS")) ret = atoi(p);
+   if(ret<0) ret=0;
+   openblas_env_openblas_num_threads=ret;
+ 
++  ret=0;
++  if (readenv(p,"OPENBLAS_NUM_THREADS")) ret = atoi(p);
++  if(ret<0) ret=0;
++  if(ret != 0 || openblas_env_openblas_num_threads == 0)
++    openblas_env_openblas_num_threads=ret;
++
+   ret=0;
+   if (readenv(p,"GOTO_NUM_THREADS")) ret = atoi(p);
+   if(ret<0) ret=0;
diff --git a/stdlib/OpenBLAS_jll/Project.toml b/stdlib/OpenBLAS_jll/Project.toml
index f04e3491ad22b..d2b76c71fe539 100644
--- a/stdlib/OpenBLAS_jll/Project.toml
+++ b/stdlib/OpenBLAS_jll/Project.toml
@@ -1,6 +1,6 @@
 name = "OpenBLAS_jll"
 uuid = "4536629a-c528-5b80-bd46-f80d51c5b363"
-version = "0.3.21+0"
+version = "0.3.21+3"
 
 [deps]
 CompilerSupportLibraries_jll = "e66e0078-7015-5450-92f7-15fbd957f2ae"
diff --git a/stdlib/OpenBLAS_jll/src/OpenBLAS_jll.jl b/stdlib/OpenBLAS_jll/src/OpenBLAS_jll.jl
index f656621d957d6..27b8f3539210a 100644
--- a/stdlib/OpenBLAS_jll/src/OpenBLAS_jll.jl
+++ b/stdlib/OpenBLAS_jll/src/OpenBLAS_jll.jl
@@ -37,6 +37,11 @@ function __init__()
         ENV["OPENBLAS_MAIN_FREE"] = "1"
     end
 
+    # Ensure that OpenBLAS does not automatically start up a bunch of threads,
+    # it will only do so if requested by another environment variable, or when
+    # LinearAlgebra initializes in its `__init__()` method:
+    ENV["OPENBLAS_DEFAULT_NUM_THREADS"] = "1"
+
     global libopenblas_handle = dlopen(libopenblas)
     global libopenblas_path = dlpath(libopenblas_handle)
     global artifact_dir = dirname(Sys.BINDIR)
